src/events/slices/events.ts

```diff
@@ -0,0 +1,118 @@
+  const events = await getListEvents();
+  return events;
+});
+
+export const fetchImageEvents = createAsyncThunk('events/fetchImageEvents', async () => {
+  const events = await getImageEvents();
+  return events;
+});
+
+export const fetchEventsByMonth = createAsyncThunk('events/fetchByMonth', async (month: DateTime) => {
+  const events = await getCalendarEvents(month);
+  return events;
+});
+
+interface IState {
+  loading: 'idle' | 'pending';
+  error: SerializedError | null;
+  entities: Record<number, IEvent>;
+}
+
+const INITIAL_STATE: IState = {
+  loading: 'idle',
+  error: null,
+  entities: {},
+};
+
+export const eventsSlice = createSlice({
+  name: 'events',
+  initialState: eventsAdapter.getInitialState(INITIAL_STATE),
+  reducers: {},
+  extraReducers: (builder) => {
+    builder.addCase(fetchEvents.pending, (state) => {
+      state.loading = 'pending';
+    }),
+      builder.addCase(fetchEvents.fulfilled, (state, action) => {
+        state.loading = 'idle';
+        eventsAdapter.addMany(state, action.payload);
+      }),
+      builder.addCase(fetchEvents.rejected, (state, action) => {
+        state.loading = 'idle';
+        state.error = action.error;
+      }),
+      builder.addCase(fetchEventById.pending, (state) => {
+        state.loading = 'pending';
+      }),
+      builder.addCase(fetchEventById.fulfilled, (state, action) => {
+        state.loading = 'idle';
+        eventsAdapter.addOne(state, action.payload);
+      }),
+      builder.addCase(fetchEventById.rejected, (state, action) => {
+        state.loading = 'idle';
+        state.error = action.error;
+      }),
+      builder.addCase(fetchEventList.pending, (state) => {
+        state.loading = 'pending';
+      }),
+      builder.addCase(fetchEventList.fulfilled, (state, action) => {
+        state.loading = 'idle';
+        eventsAdapter.addMany(state, action.payload);
+      }),
+      builder.addCase(fetchEventList.rejected, (state, action) => {
+        state.loading = 'idle';
+        state.error = action.error;
+      }),
+      builder.addCase(fetchImageEvents.pending, (state) => {
+        state.loading = 'idle';
+      }),
+      builder.addCase(fetchImageEvents.fulfilled, (state, action) => {
+        state.loading = 'idle';
+        eventsAdapter.addMany(state, action.payload);
+      }),
+      builder.addCase(fetchImageEvents.rejected, (state, action) => {
+        state.loading = 'idle';
+        state.error = action.error;
+      }),
+      builder.addCase(fetchEventsByMonth.pending, (state) => {
+        state.loading = 'idle';
+      }),
+      builder.addCase(fetchEventsByMonth.fulfilled, (state, action) => {
+        state.loading = 'idle';
+        eventsAdapter.addMany(state, action.payload);
+      }),
+      builder.addCase(fetchEventsByMonth.rejected, (state, action) => {
+        state.loading = 'idle';
+        state.error = action.error;
+      });
+  },
+});
```

Comment:
This really feels repetative! 😅 
I wonder if it's possible to merge their handling, but keep the action names?

Reply:
I have an idea that might work:
Move the api abstraction to from the API functions to thunks:
We have functions like `getListEvents`, `getImageEvents` and `getCalendarEvents` that all are an abstraction of the generic `getEvents`.

If we instead have a generic `fetchEvents` thunk which calls `getEvents`. This thunk can be used in the reducer and add multiple events like the current `fetchEvents`.
If we then have the `fetchEventsByMonth`, `fetchImageEvents` and `fetchEventList` be an abstraction of that like so:

``` typescript
// Just an example, don't know if it works
export const fetchEventList = createAsyncThunk('events/fetchList', async (_, { dispatch ) => {
  const response = await (dispatch(fetchEvents({
    event_end__gte: DateTime.local().toISODate(),
    page_size: 10,
  }));
  return unwrapResult(response);
});
```

That way we have named thunks which do pre-defined things, but only a single thunk we can abstract to our reducer

Reply:
This works like a charm! The actions becomes a bit harder to read in the devtools, as there is a lot of fetch actions, but this makes way more sense to me than to repeat the reducer methods for every single of fetch request!